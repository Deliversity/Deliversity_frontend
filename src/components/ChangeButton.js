import React, {Component} from 'react';
import {StyleSheet, View, Text, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {requestChangeUser} from '../store/actions/action';
import {connect} from 'react-redux';
import {getUserStorage, setUserStorage} from '../store/actions/action';
//배달원 혹은 사용자로 탭 전환하는 버튼
class ChangeButton extends Component {
  constructor(props) {
    super(props);
  }

  geo = () => {
    Geolocation.getCurrentPosition((position) => {
      axios.post('/api/v1/myinfo/currentLocation', position).then(() => true);
    });
  };

  async startPostPosition() {
    const timerId = setInterval(this.geo, 5 * 60000);
    await setUserStorage('timerId', timerId.toString());
  }

  async stopPostPosition() {
    const timerId = parseInt(await getUserStorage('timerId'));
    clearInterval(timerId);
  }

  onClickChange = async () => {
    // console.log(this.props.user);
    if (this.props.user === '사용자') {
      if (this.props.grade !== 2) {
        alert('등업 신청을 하세요!');
      } else {
        await this.props.requestChangeUser('배달원');
      }
    } else {
      await this.props.requestChangeUser('사용자');
    }
  };
  render() {
    return (
      <View style={styles.profile}>
        <TouchableOpacity
          onPress={() => {
            this.onClickChange();
            this.props.onPress();
          }}>
          <Icon name="bubble-chart" color="#f4da6c" size={35} />
          <Text style={styles.text_header}>{this.props.user}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  profile: {
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: 'center',
  },
  text_header: {
    color: '#ffff',
    fontWeight: 'bold',
    fontSize: 11,
  },
});
const mapStateToProps = (state) => ({
  user: state.authentication.user,
  grade: state.authentication.grade,
});
const mapDispatchToProps = (dispatch) => ({
  requestChangeUser: (data) => dispatch(requestChangeUser(data)),
});
export default connect(mapStateToProps, mapDispatchToProps)(ChangeButton);
