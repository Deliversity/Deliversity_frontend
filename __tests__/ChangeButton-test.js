import 'react-native';

import React, {ReactElement} from 'react';
import ChangeButton from '../src/components/ChangeButton';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
const mockStore = configureMockStore();
let props;
let component: ReactElement;
configure({adapter: new Adapter()});
beforeAll(() => {
  jest.mock('@react-native-community/async-storage');
  // jest.mock('@react-native-firebase');
});

describe('[Temp] render', () => {
  let store;
  props = {
    onPress: jest.fn(),
  }; // fill test props
  beforeEach(() => {
    const initialState = {
      authentication: {user: '사용자'},
    };
    store = mockStore(initialState);
    component = shallow(
      <Provider store={store}>
        <ChangeButton {...props} />
      </Provider>,
    );
  });
  it('renders without crashing', async () => {
    const renderResult = renderer.create(component);
    const renderedJson = renderResult.toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
  it('button press', () => {
    const renderResult = renderer.create(component);
    const button = renderResult.root.findByProps({testID: 'MyButton'}).props;
    renderer.act(() => button.onPress());
    const text = renderResult.root.findByProps({testID: 'MyText'}).props;
    expect(text.children).toEqual('사용자');
  });
});
