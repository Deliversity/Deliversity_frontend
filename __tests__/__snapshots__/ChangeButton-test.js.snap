// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "textAlign": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="MyButton"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#f4da6c",
            "fontSize": 35,
          },
          undefined,
          Object {
            "fontFamily": "MaterialIcons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#ffff",
          "fontSize": 11,
          "fontWeight": "bold",
        }
      }
      testID="MyText"
    >
      사용자
    </Text>
  </View>
</View>
`;
